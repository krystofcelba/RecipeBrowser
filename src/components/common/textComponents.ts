import { Text as RNText } from 'react-native';
import namedGlamorous from './namedGlamorous';

import AppTheme from 'src/assets/appTheme';

export const Text = namedGlamorous(RNText, 'Text')({
  fontFamily: 'Avenir',
  fontSize: AppTheme.defaultFontSize,
});

export const BoldText = namedGlamorous(Text, 'BoldText')({
  fontWeight: 'bold',
});

export const LeftPaddedText = namedGlamorous(Text, 'LeftPaddedText')({
  paddingLeft: AppTheme.defaultPadding / 2,
});

export const RightPaddedText = namedGlamorous(Text, 'RightPaddedText')({
  paddingRight: AppTheme.defaultPadding / 2,
});

export const LeftPaddedBoldText = namedGlamorous(LeftPaddedText, 'LeftPaddedBoldText')({
  fontWeight: 'bold',
});

export const RightPaddedBoldText = namedGlamorous(Text, 'RightPaddedBoldText')({
  fontWeight: 'bold',
});

export const TitleText = namedGlamorous(Text, 'TitleText')({
  fontSize: AppTheme.titleFontSize,
});

export const SubTitleText = namedGlamorous(BoldText, 'SubTitleText')({
  fontSize: AppTheme.subTitleFontSize,
});

export const LeftPaddedErrorText = namedGlamorous(LeftPaddedText, 'ErrorText')({
  fontSize: AppTheme.errorFontSize,
  color: AppTheme.errorTextColor,
});

export const LeftPaddedBoldErrorText = namedGlamorous(LeftPaddedBoldText, 'LeftPaddedBoldErrorText')({
  fontSize: AppTheme.errorFontSize,
  color: AppTheme.errorTextColor,
});
