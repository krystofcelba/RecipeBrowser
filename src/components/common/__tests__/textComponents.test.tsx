import React from 'react';
import { shallow } from 'enzyme';
import { Text, BoldText, LeftPaddedText, TitleText, SubTitleText } from '../textComponents';

describe('Text components', () => {
  const text = 'text';
  it('Text renders as expected', () => {
    const wrapper = shallow(<Text>{text}</Text>);
    expect(wrapper).toMatchSnapshot();
  });

  it('BoldText renders as expected', () => {
    const wrapper = shallow(<BoldText>{text}</BoldText>);
    expect(wrapper).toMatchSnapshot();
  });

  it('LeftPaddedText renders as expected', () => {
    const wrapper = shallow(<LeftPaddedText>{text}</LeftPaddedText>);
    expect(wrapper).toMatchSnapshot();
  });

  it('TitleText renders as expected', () => {
    const wrapper = shallow(<TitleText>{text}</TitleText>);
    expect(wrapper).toMatchSnapshot();
  });

  it('SubTitleText renders as expected', () => {
    const wrapper = shallow(<SubTitleText>{text}</SubTitleText>);
    expect(wrapper).toMatchSnapshot();
  });
});
