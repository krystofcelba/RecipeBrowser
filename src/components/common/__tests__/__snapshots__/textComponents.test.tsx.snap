// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text components BoldText renders as expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
      Object {
        "fontWeight": "bold",
      },
    ]
  }
>
  text
</Text>
`;

exports[`Text components LeftPaddedText renders as expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
      Object {
        "paddingLeft": 5,
      },
    ]
  }
>
  text
</Text>
`;

exports[`Text components SubTitleText renders as expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
      Object {
        "fontWeight": "bold",
      },
      Object {
        "fontSize": 20,
      },
    ]
  }
>
  text
</Text>
`;

exports[`Text components Text renders as expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
    ]
  }
>
  text
</Text>
`;

exports[`Text components TitleText renders as expected 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  disabled={false}
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "Avenir",
        "fontSize": 14,
      },
      Object {
        "fontSize": 32,
      },
    ]
  }
>
  text
</Text>
`;
